services/sidecar/src/simcore_service_sidecar/mpi_lock.py

```diff
@@ -93,9 +96,9 @@ def is_locked_factory():
 
     if not is_lock_free:
         # it was not possible to acquire the lock
-        return False
+        return
 
-    def try_to_acquire_lock_factory():
+    def try_to_acquire_lock_factory() -> bool:
         return try_to_acquire_lock(lock_manager, resource_name)
 
     # lock is free try to acquire and start background extention
```

Comment:
the return value of ``retry_for_result`` is the same as the callable ``coroutine_factory``. 
For ``coroutine_factory=try_to_acquire_lock_factory`` , it can return ``Optional[Tuple[bool, Lock]]`` so if it returns None, this will raise an exception!

Reply:
this logic was removed and replaced with a more simplified one.